lib/tasks/exercise_tests_runner.rb

```diff
@@ -0,0 +1,33 @@
+require "rake/file_utils_ext"
+require 'tmpdir'
+
+class ExerciseTestsRunner
+  include Rake::FileUtilsExt
+
+  def initialize(exercise:, test_options: '')
+    @exercise = exercise
+    @test_options = test_options
+  end
+
+  def run
+    puts "\n\n#{'-'*64}\nrunning tests for: #{exercise}"
+
+    Dir.mktmpdir(exercise.name) do |dir|
+      setup_exercise_files_in(dir)
+      run_exercise_tests_in(dir)
+    end
+  end
+
+  private
+
+  attr_reader :exercise, :test_options
+
+  def setup_exercise_files_in(dir)
+    FileUtils.cp_r exercise.directory, dir
+    FileUtils.mv "#{dir}/#{exercise.example_file}", "#{dir}/#{exercise.testable_example_file}"
+  end
+
+  def run_exercise_tests_in(dir)
+    ruby "-I lib -r disable_skip.rb #{dir}/#{exercise.test_file} #{test_options}"
```

Comment:
You want to mock `ruby` and assert that it gets called with the right options.

When you go down that road you end up extracting several methods/objects that purely exist to aid testing. (I don't think this is a bad thing, but that is not a generally popular opinion, but most people are **really** bad at testing.)

1. test that the arguments to the ruby command are as expected. (extract a method that composes them, this is easily tested by string comparison.)
2. test that the `ruby` command is called with some arguments (because of 1 we don't care what they are, so if they are `-e 'p true'` it doesn't really matter, and when you get to THAT stage you begin to question whether mocking `ruby` is worth the effort or if you should just call it anyway.)

Note: that `ruby` here is a rake FileUtils method: http://rake.rubyforge.org/FileUtils.html#method-i-ruby